libia2/include/ia2_compartment_init.inc

```diff
@@ -60,3 +65,135 @@ void COMPARTMENT_IDENT(init_tls)(void) {
   };
   dl_iterate_phdr(protect_tls_pages, &args);
 }
+
+/* Magic symbol from the linker that marks the start of the ELF headers */
+extern Elf64_Ehdr __ehdr_start __attribute__((weak));
```

Comment:
What compartments can write to `__ehdr_start`? I'm assuming it can't be relro since it's defined by the linker right?